dex/src/state.rs

```diff
@@ -2422,10 +2429,6 @@ impl State {
                         }
                         _ => (),
                     };
-                    if !maker {
-                        let referrer_rebate = fees::referrer_rebate(native_fee_or_rebate);
-                        open_orders.referrer_rebates_accrued += referrer_rebate;
-                    }
```

Comment:
I think that if this is deployed in a program upgrade, any rebates sitting in the event queues will be lost.

Reply:
Yes, you're right. So the total loss would be the 20% of the fee value of all trades that are on the event queue at the time this deployment goes out. My instinct is that it's a small loss because cranks usually clear out the event queue very quickly and most queues don't have trades on them, but I don't actually know the stats on this.

It simplifies the settling process for us by a lot. With the referrer rebate being granted at time of trade, we can put a `SettleFunds` instruction right after a `NewOrder` instruction and not have to wait to send a settle funds. We try to hide the settling thing as much as possible from users :smile:

Reply:
My guesstimate is that both the UX improvement and the 20k compute units saved are worth more than the 20% of the fee value of all trades at time of the deployment.

Reply:
20k compute units :open_mouth: ? Spl token transfer CPI usually is ~3k units.
But even 3k units saving is good.